### PR TITLE
MINOR: AdminClient should register with `AppInfoParser`

### PR DESCRIPTION
**Pull request by ijuma**
Also make "created" message more consistent across clients.

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @hachikuji. We should cherry-pick to 0.11.0 as well.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk7-scala2.11/6706/
Test PASSed (JDK 7 and Scala 2.11).

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/kafka-pr-jdk8-scala2.12/6691/
Test PASSed (JDK 8 and Scala 2.12).

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java [hunk 1]
```diff
@@ -388,6 +390,9 @@ public void close(long duration, TimeUnit unit) {
         try {
             // Wait for the thread to be joined.
             thread.join();
+
+            AppInfoParser.unregisterAppInfo(JMX_PREFIX, clientId);
+            
             log.debug("{}: closed.", clientId);
```
Comment (omkreddy) [inline clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java:396]: nit: debug statement differs from other clients

**Comment by omkreddy** [comment]
+1

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java (hunk 1 above)
Comment (ijuma) [reply clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java:396]: Yes, I know, but the other messages in the `close` statement follow this pattern. So, it seemed more important to be consistent with the other log messages in this class than other classes.

**Comment by ijuma** [comment]
cc @cmccabe

**Review by hachikuji — APPROVED** [review]
LGTM

**Comment by ijuma** [comment]
Merged to trunk and 0.11.0 branches.

**Pull request closed without merging**